clang/lib/Sema/SemaExpr.cpp

```diff
@@ -13736,6 +13756,42 @@ QualType Sema::CheckAssignmentOperands(Expr *LHSExpr, ExprResult &RHS,
     ConvTy = CheckAssignmentConstraints(Loc, LHSType, RHSType);
   }
 
+  // __builtin_counted_by_ref can't be used in a binary expression or array
+  // subscript on the LHS.
+  int DiagOption = -1;
+  auto FindInvalidUseOfBoundsSafetyCounter = [&](Expr *E) {
```

Comment:
Same here. I couldn't find a check to find the use of `__builtin_counted_by_ref`. Wasn't it to find things like below?

```
__builtin_counted_by_ref(something)[1] = 0;
*(__builtin_counted_by_ref(something) + 1) = 0;
```

Reply:
It's the same issue where `__builtin_counted_by_ref` isn't in the AST anymore.